### PR TITLE
Fix SignTool IsManaged check and do strong name check on tests

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: #1876 
Closes: #1871 

- Fix to correctly detect if a file is managed or not. It just checks for the existence of the Cor20 (CLR) Header. Also add try/catch for handling non-PE files.
- Enable strong-name check by default for the unit tests.
- Added additional inputs to cross-gen test case.
- The cross-gen hadn't a good coverage because the .DLLs were being signed by default and strong name _check_ wasn't being performed during tests.

PTAL

### CODE REVIEW

File: src/Microsoft.DotNet.SignTool/src/ContentUtil.cs [hunk 1]
```diff
@@ -69,12 +69,17 @@ public static bool IsPublicSigned(PEReader peReader)
 
         public static bool IsManaged(string filePath)
         {
-            using (var stream = new FileStream(filePath, FileMode.Open))
-            using (var peReader = new PEReader(stream))
+            try
```
Comment (weshaggard) [inline src/Microsoft.DotNet.SignTool/src/ContentUtil.cs]: This obviously conflicts with my PR. I don't have any problem if you want to close my PR in favor of yours.

File: src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs [hunk 2]
```diff
@@ -194,7 +194,7 @@ public void Dispose()
             var signingInput = new Configuration(signToolArgs.TempDir, itemsToSign, strongNameSignInfo, fileSignInfo, extensionsSignInfo, dualCertificates, task.Log).GenerateListOfFiles();
             var util = new BatchSignUtil(task.BuildEngine, task.Log, signTool, signingInput, new string[] { });
 
-            util.Go(false);
+            util.Go(true);
```
Comment (natemcmaster) [inline src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs]: What does this do? The API isn't very descriptive, so not sure if this change is right or not.

Comment (jcagme) [reply src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs]: 👍 I was about to say the same thing

Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs]: It enables strong-name check by default for the unit tests. I'll add a label to the parameter.

**Comment by JohnTortugo** [comment]
> Can you please verify that the new test fails without the fix to IsManaged? Otherwise the changes look good.

Yeah, I could repro these problems locally. Also, enabling the strong-name check failed a few test cases before I implemented the fix.

**Pull request merged**